packages/asset-swapper/src/utils/market_operation_utils/bancor_service.ts

```diff
@@ -31,29 +31,44 @@ export class BancorService {
         return this._sdk;
     }
 
-    public async getQuoteAsync(fromToken: string, toToken: string, amount: BigNumber): Promise<Quote<BancorFillData>> {
+    public async getQuotesAsync(
+        fromToken: string,
+        toToken: string,
+        amounts: BigNumber[],
+    ): Promise<Array<Quote<BancorFillData>>> {
         const sdk = await this.getSDKAsync();
         const blockchain = sdk._core.blockchains[BlockchainType.Ethereum] as Ethereum;
         const sourceDecimals = await getDecimals(blockchain, fromToken);
-        const { path, rate } = await sdk.pricing.getPathAndRate(
+        const quotes = await sdk.pricing.getPathAndRates(
```

Comment:
How many outgoing requests this makes?

We can add logging to RPCSubprovider (or use the add a DebugSubprovider before RPC) to investigate.